src/sentry/integrations/msteams/webhook.py

```diff
@@ -243,25 +246,25 @@ class MsTeamsWebhookEndpoint(Endpoint, MsTeamsWebhookMixin):
 
     def __init__(self, **kwargs) -> None:
         super().__init__(**kwargs)
-        self._event_handlers: dict[MsTeamsEvents, Callable[[HttpRequest], HttpResponse]] = {
-            MsTeamsEvents.MESSAGE: self.handle_message_event,
-            MsTeamsEvents.CONVERSATION_UPDATE: self.handle_conversation_update_event,
-            MsTeamsEvents.INSTALLATION_UPDATE: self.handle_installation_update_event,
-            MsTeamsEvents.UNKNOWN: self.handle_unknown_event,
+        self._event_handlers: dict[MsTeamsEvents, Callable[[Request], Response]] = {
+            MsTeamsEvents.MESSAGE: self._handle_message_event,
+            MsTeamsEvents.CONVERSATION_UPDATE: self._handle_conversation_update_event,
+            MsTeamsEvents.INSTALLATION_UPDATE: self._handle_installation_update_event,
+            MsTeamsEvents.UNKNOWN: self._handle_unknown_event,
         }
 
     @csrf_exempt
-    def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+    def dispatch(self, request: Request, *args, **kwargs) -> Response:  # type: ignore[override]
```

Comment:
I could use some help understanding the error that this "type:ignore" is suppressing, which is:

```
error: Argument 1 of "dispatch" is incompatible with supertype "APIView"; supertype defines the argument type as "HttpRequest"  [override]
```

`sentry.api.base.Endpoint.dispatch` has the parameter type `rest_framework.requests.Request`, which is also what this class gets at runtime and consistent with the rest of the class wanting to read `request.data` everywhere. The error message suggests that it's the base `Endpoint` class that's incorrectly overriding `APIView`, but AFAICT that class is covered by type-checking and produces no error.

Furthermore, `APIView` itself doesn't have any type annotations, at least not in the source code that's showing up in my venv. And it's a `rest_framework` class too, so I don't know why it wouldn't use the `rest_framework` request class.

For context, `HttpRequest` has no `data` attribute and is wrong to use in the rest of this class's handler methods. The problem does not appear to be unique to this class; I'm seeing `def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse` in a few other places, but not consistently.

Reply:
our view class hierarchy is a disaster -- it's kinda impossible to know what is and isn't a rest framework view at the moment

at some point these need to be fixed from the bottom up -- and perhaps without inheritance -- but it's such a large problem that I haven't even started thinking about how to approach it

technically even for `rest_framework` `def dispatch` still takes the vanilla django request -- rest_frakework then converts it into its special class.  so the type checker is actually correct here that this should be `(HttpRequest, *object, **object) -> HttpResponseBase`

Reply:
Glad to know it's not just me. 😅

> technically even for `rest_framework` `def dispatch` still takes the vanilla django request -- rest_frakework then converts it into its special class. so the type checker is actually correct here that this should be `(HttpRequest, *object, **object) -> HttpResponseBase`

That makes sense. I took another pass and it turns out that Mypy is cool if I leave this as `dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse` and change `post` and everything else to use `rest_framework.Request`. I had thought it would complain about the `super().dispatch` call (PyCharm even highlights it with a warning) but I guess not.